api/envoy/extensions/filters/http/alternate_protocols_cache/v3/alternate_protocols_cache.proto

```diff
@@ -17,9 +18,8 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 // Configuration for the alternate protocols cache HTTP filter.
 // [#extension: envoy.filters.http.alternate_protocols_cache]
 message FilterConfig {
-  // If set, causes the use of the alternate protocols cache, which is responsible for
-  // parsing and caching HTTP Alt-Svc headers. This enables the use of HTTP/3 for upstream
-  // servers that advertise supporting it.
-  // TODO(RyanTheOptimist): Make this field required when HTTP/3 is enabled via auto_http.
-  config.core.v3.AlternateProtocolsCacheOptions alternate_protocols_cache_options = 1;
+  // This field is ignored: the alternate protocols cache filter will use the
+  // cache for the cluster the request is routed to.
+  config.core.v3.AlternateProtocolsCacheOptions alternate_protocols_cache_options = 1
+      [deprecated = true, (envoy.annotations.deprecated_at_minor_version) = "3.0"];
```

Comment:
hmmm, others change is fine. But according our API policy, we should never remove any implementation of exist API even it's deprecated. Not sure if this is a break change to our API? cc @envoyproxy/api-shepherds 

```
Deprecations will still occur as an end-user indication that there is a preferred way to configure a particular feature, but no field will ever be removed nor will Envoy ever remove the implementation for any deprecated field.
```

(PS: I personally think this rule is too strict. Envoy need to take on more and more debt. I personally think we should loose this restriction on non-core APIs (extension API or unusual API?).)

Reply:
yeah so I think the policy is that we never remove support for a feature.  The underlying feature here is being able to cache alternate protocol. the way it was cached actually caused bugs so we deprecated the bug, but not the ability to cache alternate protocol.  I think this is aligned with our API guidelines but I'll let shepherds sign off.

Reply:
oh oops, you were the shephard, haha.  I'll tag Adi for a second opinion =P

Reply:
I agree that marking this field deprecated is ok, but changing the implementation (removing support) is against the current policy.

The current deprecation implies that the field is still usable, and this is not the case at the moment. A clear warning needs to be added for anyone using this field. I also suggest adding a link to the upstream cluster's config (not sure if it is this one: https://github.com/envoyproxy/envoy/blob/f9c14577ae2f07ccf0586de71a3d3e83deae3eb8/api/envoy/extensions/upstreams/http/v3/http_protocol_options.proto#L129).

Whether the behavior can be removed completely or not should be agreed by the rest of the @envoyproxy/api-shepherds as this contradicts:
```
NOTE: Client implementations are free to output additional warnings about field usage beyond deprecation, if for example, the continued use of the field is deemed a substantial security risk.
Individual client versions are also free to stop supporting fields if they want to, though Envoy Proxy (as an xDS client) commits to never doing so.
```

Reply:
so it used to be the case that you had to configure this in cluster, and filter, and the two configs had to match or config would fail to load.

literally all we are doing is removing the knob here and ignoring the config here.  Any prior configuration will work 100% the same as it did so no customer will be affected.

This allows previously unsupported option of having one listener with two clusters with different cache configuration.  No functionality is deprecated, no configuration is invalidated, and we've added a feature.

Reply:
> so it used to be the case that you had to configure this in cluster, and filter, and the two configs had to match or config would fail to load.

Having a single definition of config is great (ODR FTW).
What is unclear to me is what was/is the expected behavior of the following (apologies, as my knowledge on the AltSvc protocol and caching options isn't deep):
1. a request goes through a route that isn't configured with the AlternateProtocols filter, and ends up at a cluster that does have AlternateProtocols configured.
2. a request goes through a route that has AlternateProtocols filter configured, and ends up at a cluster that does not have AlternateProtocols configured.

If the same behavior is kept in all cases, then I think it is ok to remove the implementation of the field.

Reply:
previously the config (2) would be rejected - you couldn't have a alternate protocols filter and a cluster whose config didn't match.

Reply:
Apologies for the late reply. If that's the case, I think this change is ok.
I still think there should be a clear warning to users of this field, so it will be clear that their current Envoy build is disregarding the field.

Reply:
like log warning?  We already whine when using deprecated fields AFIK but I can add a second warning

Reply:
IIUC in this case the field is essentially removed, and not deprecated (which implies that it is still implemented), and therefore I think it should be clarified with a warning.

Thanks for [adding the log line](https://github.com/envoyproxy/envoy/pull/33730/commits/ceaed37bae3bcaf503960221484b98e6d7c416b0). Off to @wbpcode for final review.

Reply:
I think we can even add a `[#not-implemented-hide:]` to hide this field in the docs.